umpost/conversion_driver_esm1p5.py

```diff
@@ -0,0 +1,201 @@
+    # See get_name.F90 in the UM7.3 source code for details.
+    fields_file_name_pattern = rf"^{run_id}a.p[a-z0-9]+$"
+
+    return fields_file_name_pattern
+
+
+def set_nc_write_path(fields_file_path, nc_write_dir):
+    """
+    Set filepath for writing NetCDF to based on fields file name.
+
+    Parameters
+    ----------
+    fields_file_path : path to single UM fields file to be converted.
+    nc_write_dir : path to target directory for writing NetCDF files.
+
+    Returns
+    -------
+    nc_write_path : path for writing converted fields_file_path file.
+    """
+    fields_file_path = (
+        Path(fields_file_path)
+        if isinstance(fields_file_path, str)
+        else fields_file_path
+    )
+
+    nc_write_dir = Path(nc_write_dir) if isinstance(nc_write_dir, str) else nc_write_dir
+
+    fields_file_name = fields_file_path.name
+    nc_name = fields_file_name + ".nc"
+    nc_write_path = nc_write_dir / nc_name
+
+    return nc_write_path
+
+
+def find_matching_fields_files(dir_contents, fields_file_name_pattern):
+    """
+    Find files in list of paths with names matching fields_file_name_pattern.
+    Used to find ESM1.5 UM outputs in a simulation output directory.
+
+    Parameters
+    ----------
+    dir_contents : list of file paths, typically contents of a single directory.
+    fields_file_name_pattern : Regex pattern for matching fields file names.
+
+    Returns
+    -------
+    fields_file_paths : subset of dir_contents with names matching fields_file_name_pattern.
+    """
+
+    fields_file_paths = []
+    for filepath in dir_contents:
+        filepath = Path(filepath)
+
+        if re.match(fields_file_name_pattern, filepath.name):
+            fields_file_paths.append(filepath)
+
+    return fields_file_paths
+
+
+def convert_fields_file_list(fields_file_path_list, nc_write_dir):
+    """
+    Convert group of fields files to NetCDF, writing output in nc_write_dir.
+
+    Parameters
+    ----------
+    fields_file_list : list of paths to fields files for conversion.
+    nc_write_dir : directory to save NetCDF files into.
+
+    Returns
+    -------
+    None
+    """
+
+    for fields_file_path in fields_file_path_list:
+
+        fields_file_path = (
+            Path(fields_file_path)
+            if isinstance(fields_file_path, str)
+            else fields_file_path
+        )
+
+        nc_write_path = set_nc_write_path(fields_file_path, nc_write_dir)
+
+        print("Converting file " + fields_file_path.name)
+
+        try:
+            um2netcdf4.process(fields_file_path, nc_write_path, ARG_VALS)
+
+        except Exception as exc:
+            # Not ideal here - um2netcdf4 raises generic exception when missing coordinates
+            if exc.args[0] in ALLOWED_UM2NC_EXCEPTION_MESSAGES.values():
+                warnings.warn("Unable to convert file: " + fields_file_path.name)
+            else:
+                raise
+
+
+def convert_esm1p5_output_dir(current_output_dir):
```

Comment:
I've got linter warning about variable name shadowing with `current_output_dir`.

Reply:
Just renamed it to `esm1p5_output_dir` and left the global one as is.